examples/bank/bank.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2014 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Vivek Menezes (vivek.menezes@gmail.com)
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/security/securitytest"
+	"github.com/cockroachdb/cockroach/server"
+	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/testutils"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// makes an id string from an id int.
+func makeAccountID(id int) []byte {
+	return []byte(fmt.Sprintf("%09d", id))
+}
+
+// Bank stores all the bank related state.
+type Bank struct {
+	servingAddr string
+	numAccounts int
+	// This is not protected, but we really don't care.
+	approxNumTransactions int
+}
+
+// moveMoney() moves an amount between two accounts if the amount
+// is available in the from account. Returns true on success.
+func (bank *Bank) moveMoney(kvClient *client.KV, from, to []byte, amount int64) bool {
+	txnOpts := &client.TransactionOptions{Name: fmt.Sprintf("Transferring %d", amount)}
+	err := kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		call := client.Get(proto.Key(from))
+		gr := call.Reply.(*proto.GetResponse)
+		if err := txn.Run(call); err != nil {
+			return err
+		}
+		// Read from value.
+		var fromValue int64
+		if gr.Value != nil && gr.Value.Bytes != nil {
+			readValue, err := strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+			if err != nil {
+				return err
+			}
+			fromValue = readValue
+		}
+
+		// Ensure there is enough cash.
+		if fromValue < amount {
+			return nil
+		}
+
+		// Read to value.
+		call = client.Get(proto.Key(to))
+		gr = call.Reply.(*proto.GetResponse)
+		if err := txn.Run(call); err != nil {
```

Comment:
You should read the from and to accounts in parallel. You do this by create 2 calls:

```
  get1 := client.Get(proto.Key(from))
  get2 := client.Get(proto.Key(to))
  if err := txn.Run(get1, get2); err != nil {
    return err
  }
```